pkg/collector/python/version.go

```diff
@@ -0,0 +1,35 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package python
+
+import (
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/util/cache"
+)
+
+var (
+	pythonInfoCacheKey = cache.BuildAgentKey("pythonInfo")
+)
+
+// GetPythonInfo returns the info string as provided by the embedded Python interpreter.
+//
+// Example: '3.10.6 (main, May 29 2023, 11:10:38) [GCC 11.3.0]'
+func GetPythonInfo() string {
+	// retrieve the Python version from the Agent cache
+	if x, found := cache.Cache.Get(pythonInfoCacheKey); found {
+		return x.(string)
+	}
+
+	return "n/a"
+}
+
+// GetPythonVersion returns the embedded python version as provided by the embedded Python interpreter.
+//
+// Example: '3.10.6'
+func GetPythonVersion() string {
+	return strings.Split(GetPythonInfo(), " ")[0]
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 tiniest nitpick:** ‏there is a `strings.Cut` function which would probably be cleaner than `Split` then getting the first element of the array, as it returns directly the left and right part of the cut

[once again if you don't want to do any change in the implementation just ignore this comment]

Reply:
Good catch, I move the code without really looking at it.

Reply:
My thought was more to avoid the index access (if the string is empty it would return an empty slice and crash)
The doc of `Split` even says `To split around the first instance of a separator, see Cut.`
But in any case the string shouldn't be empty so feel free to keep it that way if you don't want to change the implementation, definitely not something critical